### PR TITLE
feat: temporal client - configuration helpers

### PR DESCRIPTION
**Pull request by itayd**
allows standardized temporal configuration across the system. will be used in a later pr.

### CODE REVIEW

File: internal/kittehs/choose.go [hunk 1]
```diff
@@ -0,0 +1,12 @@
+package kittehs
+
+// Return first opt which is non-zero.
```
Comment (daabr) [inline internal/kittehs/choose.go]: 1. https://go.dev/doc/comment#func
2. Add unit test(s) for this function

File: internal/backend/temporalclient/worker.go [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+}
+
+// other overrides self.
+func (wc WorkerConfig) With(other WorkerConfig) WorkerConfig {
+	return WorkerConfig{
+		WorkflowDeadlockTimeout: kittehs.Choose(other.WorkflowDeadlockTimeout, wc.WorkflowDeadlockTimeout),
+	}
+}
+
+func NewWorker(l *zap.Logger, client client.Client, qname string, cfg WorkerConfig) worker.Worker {
```
Comment (itayd) [inline internal/backend/temporalclient/worker.go:29]: document why

Comment (itayd) [reply internal/backend/temporalclient/worker.go:29]: add default

File: internal/backend/temporalclient/slow.go [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	"go.temporal.io/sdk/workflow"
+)
+
+func Slow[T any](
```
Comment (itayd) [inline internal/backend/temporalclient/slow.go]: add doc

Comment (itayd) [reply internal/backend/temporalclient/slow.go]: change name

File: internal/kittehs/choose.go (hunk 1 above)
Comment (itayd) [reply internal/kittehs/choose.go]: done

File: internal/backend/temporalclient/worker.go (hunk 2 above)
Comment (itayd) [reply internal/backend/temporalclient/worker.go:29]: done

File: internal/backend/temporalclient/slow.go (hunk 3 above)
Comment (itayd) [reply internal/backend/temporalclient/slow.go]: done

**Review by daabr — APPROVED** [review]

**Pull request merged**